src/utils.c

```diff
@@ -7,6 +7,19 @@
 #include <stdlib.h>
 #include <string.h>
 
+char* string_cut_maxlen(char* string, int maxlen){
+        char *buffer;
+
+        if (maxlen < 0 || strlen(string) < maxlen) {
+                buffer = string;
+        } else {
+                size_t buffer_maxlen = maxlen;
+                buffer = calloc(buffer_maxlen, sizeof(char));
```

Comment:
We're using glib so instead of `calloc` `g_malloc0` should be used but since we already know we are going to fill the entire buffer there is no reason to zero it out so `g_malloc` would probably be the correct function to use here.